src/functions-reference/matrix_operations.Rmd

```diff
@@ -1237,66 +1237,270 @@ The cumulative sum of v
-function that builds a $K \in \mathbb{R}^{N \times M}$ covariance
-matrix for $x_1, \dots, x_N$ and $x^\prime_1, \dots, x^\prime_M$,
-where $x_i \in \mathbb{R}^D$ and $x^\prime_i \in \mathbb{R}^D$ and
-$K_{i,j} = k(x_i, x^\prime_j)$.
+With scale $\sigma$ and length scale $l$, the squared exponential kernel is:
 
-<!-- matrix; cov_exp_quad; (row_vectors x, real alpha, real rho); -->
-\index{{\tt \bfseries cov\_exp\_quad }!{\tt (row\_vectors x, real alpha, real rho): matrix}|hyperpage}
+$$
+k(x_i, x_j) = \alpha^2 \exp \left( -\dfrac{1}{2\rho^2} |x_i - x_j|^2 \right)
+$$
 
-`matrix` **`cov_exp_quad`**`(row_vectors x, real alpha, real rho)`<br>\newline
-The covariance matrix with an exponentiated quadratic kernel of x.
+\index{{\tt \bfseries gp\_exp\_quad\_cov }!{\tt (real[] x, real sigma, real length_scale): matrix}|hyperpage}
 
-<!-- matrix; cov_exp_quad; (vectors x, real alpha, real rho); -->
-\index{{\tt \bfseries cov\_exp\_quad }!{\tt (vectors x, real alpha, real rho): matrix}|hyperpage}
+`matrix` **`gp_exp_quad_cov`**`(real[] x, real sigma, real length_scale)`<br>\newline
 
-`matrix` **`cov_exp_quad`**`(vectors x, real alpha, real rho)`<br>\newline
-The covariance matrix with an exponentiated quadratic kernel of x.
+Gaussian process with squared exponential kernel in one dimension.
 
-<!-- matrix; cov_exp_quad; (real[] x, real alpha, real rho); -->
-\index{{\tt \bfseries cov\_exp\_quad }!{\tt (real[] x, real alpha, real rho): matrix}|hyperpage}
+\index{{\tt \bfseries gp\_exp\_quad\_cov }!{\tt (real[] x1, real[] x2, real sigma, real length_scale): matrix}|hyperpage}
 
-`matrix` **`cov_exp_quad`**`(real[] x, real alpha, real rho)`<br>\newline
-The covariance matrix with an exponentiated quadratic kernel of x.
+`matrix` **`gp_exp_quad_cov`**`(real[] x1, real[] x2, real sigma, real length_scale)`<br>\newline
 
-<!-- matrix; cov_exp_quad; (row_vectors x1, row_vectors x2, real alpha, real rho); -->
-\index{{\tt \bfseries cov\_exp\_quad }!{\tt (row\_vectors x1, row\_vectors x2, real alpha, real rho): matrix}|hyperpage}
+Gaussian process with squared exponential kernel in one dimension between `x1`
+and `x2`.
 
-`matrix` **`cov_exp_quad`**`(row_vectors x1, row_vectors x2, real alpha, real rho)`<br>\newline
-The covariance matrix with an exponentiated quadratic kernel of x1 and
-x2.
+\index{{\tt \bfseries gp\_exp\_quad\_cov }!{\tt (vector[] x, real sigma, real length_scale): matrix}|hyperpage}
 
-<!-- matrix; cov_exp_quad; (vectors x1, vectors x2, real alpha, real rho); -->
-\index{{\tt \bfseries cov\_exp\_quad }!{\tt (vectors x1, vectors x2, real alpha, real rho): matrix}|hyperpage}
+`matrix` **`gp_exp_quad_cov`**`(vector[] x, real sigma, real length_scale)`<br>\newline
 
-`matrix` **`cov_exp_quad`**`(vectors x1, vectors x2, real alpha, real rho)`<br>\newline
-The covariance matrix with an exponentiated quadratic kernel of x1 and
-x2.
+Gaussian process with squared exponential kernel in multiple dimensions.
```

Comment:
This is picky, and optional, but I'd be explicit about the dimension of D being greater than 1.

Reply:
I left it at multiple dimensions. The code accepts 1D, so it'd work if someone did it.